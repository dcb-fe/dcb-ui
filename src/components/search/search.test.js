import Search from './search.vue';
import { shallowMount } from '@vue/test-utils';

describe('Search', () => {
  describe('视图', () => {
    test('普通展示', () => {
      const search = shallowMount(Search);
      expect(search.html()).toMatchSnapshot();
    });

    test('自定义右侧内容', () => {
      const search = shallowMount(Search, {
        slots: {
          right: '<span class="btn">搜索</span>',
        },
      });
      expect(search.find('.btn').exists()).toBe(true);
      expect(search.html()).toMatchSnapshot();
    });

    test('圆形搜索框', () => {
      const search = shallowMount(Search, {
        props: {
          shape: 'round',
        },
      });
      expect(search.html()).toMatchSnapshot();
    });

    test('搜索框内容对齐', () => {
      const search = shallowMount(Search, {
        props: {
          inputAlign: 'center',
        },
      });
      expect(search.html()).toMatchSnapshot();
    });

    test('自定义背景色', () => {
      const search = shallowMount(Search, {
        props: {
          background: 'green',
        },
      });
      expect(search.html()).toMatchSnapshot();
    });
  });

  describe('事件', () => {
    test('确定搜索触发', () => {
      const handerSearch = jest.fn();
      const search = shallowMount(Search, {
        listeners: {
          search: handerSearch,
        },
      });
      search.vm.$emit('search');
      search.find('input').trigger('keyup', { keyCode: 13 });
      expect(handerSearch).toBeCalled().toBeCalledTimes(2);
    });

    test('输入框输入时触发', () => {
      const handerInput = jest.fn();
      const search = shallowMount(Search, {
        listeners: {
          input: handerInput,
        },
      });
      search.find('input').trigger('input');
      expect(handerInput).toBeCalled().toBeCalledTimes(1);
    });

    test('输入框聚焦时触发', () => {
      const handerFocus = jest.fn();
      const search = shallowMount(Search, {
        attachTo: 'body', // 测试focus必须有attachTo属性，详情请查看官网
        listeners: {
          focus: handerFocus,
        },
      });
      search.find('input').trigger('focus', {});
      expect(handerFocus).toBeCalled().toBeCalledTimes(1);
    });

    test('输入框失焦时触发', () => {
      const handerBlur = jest.fn();
      const search = shallowMount(Search, {
        listeners: {
          blur: handerBlur,
        },
      });
      search.find('input').trigger('blur');
      expect(handerBlur).toBeCalled().toBeCalledTimes(1);
    });
  });
});
