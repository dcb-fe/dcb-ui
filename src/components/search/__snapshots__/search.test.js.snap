// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search 视图 圆形搜索框 1`] = `
<div class="d-search" style="background-color: white;">
  <form action="javascript:void 0" class="d-search-form">
    <div class="d-search-content">
      <d-icon-stub name="search"></d-icon-stub> <input type="text" placeholder="请输入搜索关键字" class="d-search-control d-search-left">
      <d-icon-stub name="close" style="display: none;"></d-icon-stub>
    </div>
    <!---->
  </form>
</div>
`;

exports[`Search 视图 搜索框内容对齐 1`] = `
<div class="d-search" style="background-color: white;">
  <form action="javascript:void 0" class="d-search-form">
    <div class="d-search-content">
      <d-icon-stub name="search"></d-icon-stub> <input type="text" placeholder="请输入搜索关键字" class="d-search-control">
      <d-icon-stub name="close" style="display: none;"></d-icon-stub>
    </div>
    <!---->
  </form>
</div>
`;

exports[`Search 视图 普通展示 1`] = `
<div class="d-search" style="background-color: white;">
  <form action="javascript:void 0" class="d-search-form">
    <div class="d-search-content">
      <d-icon-stub name="search"></d-icon-stub> <input type="text" placeholder="请输入搜索关键字" class="d-search-control d-search-left">
      <d-icon-stub name="close" style="display: none;"></d-icon-stub>
    </div>
    <!---->
  </form>
</div>
`;

exports[`Search 视图 自定义右侧内容 1`] = `
<div class="d-search" style="background-color: white;">
  <form action="javascript:void 0" class="d-search-form">
    <div class="d-search-content">
      <d-icon-stub name="search"></d-icon-stub> <input type="text" placeholder="请输入搜索关键字" class="d-search-control d-search-left">
      <d-icon-stub name="close" style="display: none;"></d-icon-stub>
    </div>
    <div class="d-search-action"><span class="btn">搜索</span></div>
  </form>
</div>
`;

exports[`Search 视图 自定义背景色 1`] = `
<div class="d-search">
  <form action="javascript:void 0" class="d-search-form">
    <div class="d-search-content">
      <d-icon-stub name="search"></d-icon-stub> <input type="text" placeholder="请输入搜索关键字" class="d-search-control d-search-left">
      <d-icon-stub name="close" style="display: none;"></d-icon-stub>
    </div>
    <!---->
  </form>
</div>
`;
